kornia/geometry/epipolar/fundamental.py

```diff
@@ -70,9 +70,212 @@ def normalize_transformation(M: Tensor, eps: float = 1e-8) -> Tensor:
+    delta = b * b - 4 * a * c
+
+    if delta < 0:
+        return torch.tensor(0)
+
+    inv_2a = 0.5 / a
+
+    if delta == 0:
+        x1 = -b * inv_2a
+        x2 = x1
+        return x1
+
+    sqrt_delta = torch.sqrt(delta)
+    x1 = (-b + sqrt_delta) * inv_2a
+    x2 = (-b - sqrt_delta) * inv_2a
+
+    return torch.cat([x2, x1])
+
+
+def solve_cubic(a: torch.Tensor, b: torch.Tensor, c: torch.Tensor, d: torch.Tensor) -> torch.Tensor:
+    _PI = torch.tensor(3.141592653589793, device=a.device, dtype=a.dtype)
+
+    if a == 0:
+        # second order system
+        if b == 0:
+            # first order system
+            if c == 0:
+                return torch.tensor(0, device=a.device, dtype=a.dtype)
+            x0 = -d / c
+            return torch.tensor(1.0, device=a.device, dtype=a.dtype)
+        x2 = 0
+        return solve_quadratic(b, c, d)
+
+    # normalized form x^3 + a2 * x^2 + a1 * x + a0 = 0
+    inv_a = 1.0 / a
+    b_a = inv_a * b
+    b_a2 = b_a * b_a
+
+    c_a = inv_a * c
+    d_a = inv_a * d
+
+    # solve the cubic equation
+    Q = (3 * c_a - b_a2) / 9
+    R = (9 * b_a * c_a - 27 * d_a - 2 * b_a * b_a2) / 54
+    Q3 = Q * Q * Q
+    D = Q3 + R * R
+    b_a_3 = (1.0 / 3.0) * b_a
+
+    if Q == 0:
+        if R == 0:
+            x0 = x1 = x2 = -b_a_3
+            return torch.stack([x2, x1, x0])
+
+        else:
+            cube_root = torch.pow(2 * R, 1 / 3)
+            x0 = cube_root - b_a_3
+            return x0
+
+    if D <= 0:
+        # three real roots
+        theta = torch.acos(R / torch.sqrt(-Q3))
+        sqrt_Q = torch.sqrt(-Q)
+        x0 = 2 * sqrt_Q * torch.cos(theta / 3.0) - b_a_3
+        x1 = 2 * sqrt_Q * torch.cos((theta + 2 * _PI) / 3.0) - b_a_3
+        x2 = 2 * sqrt_Q * torch.cos((theta + 4 * _PI) / 3.0) - b_a_3
+        return torch.stack([x2, x1, x0])
+
+    # D > 0, only one real root
+    AD = torch.tensor(0, device=a.device, dtype=a.dtype)
+    BD = torch.tensor(0, device=a.device, dtype=a.dtype)
+    R_abs = torch.abs(R)
+    if R_abs > 1e-16:
+        AD = torch.pow(R_abs + torch.sqrt(D), 1 / 3)
+        AD = AD if R >= 0 else -AD
+        BD = -Q / AD
+    x0 = AD + BD - b_a_3
+
+    return x0
+
+
+# Reference: Adapted from the 'run_7point' function in opencv
+# https://github.com/opencv/opencv/blob/4.x/modules/calib3d/src/fundam.cpp
+def run_7point(points1: torch.Tensor, points2: torch.Tensor) -> torch.Tensor:
+    r"""Compute the fundamental matrix using the 7-point algorithm.
+
+    Args:
+        points1: A set of points in the first image with a tensor shape :math:`(B, N, 2), N==7`.
+        points2: A set of points in the second image with a tensor shape :math:`(B, N, 2), N==7`.
+
+    Returns:
+        the computed fundamental matrix with shape :math:`(B, 3*m, 3), Valid values of m are 1, 2 or 3`
+    """
+    if points1.shape != points2.shape:
+        raise AssertionError(points1.shape, points2.shape)
```

Comment:
from kornia.core.check import KORNIA_CHECK_SHAPE
```suggestion
    KORNIA_CHECK_SHAPE(points1.shape, ['B', '7', '2'])
    KORNIA_CHECK_SHAPE(points2.shape, ['B', '7', '2'])
```